providers/factories/oracle/api.go

```diff
@@ -0,0 +1,83 @@
+package oracle
+
+import (
+	"fmt"
+	"math/big"
+	"net/http"
+
+	"github.com/skip-mev/slinky/oracle/config"
+	"github.com/skip-mev/slinky/pkg/math"
+	"github.com/skip-mev/slinky/providers/apis/binance"
+	coinbaseapi "github.com/skip-mev/slinky/providers/apis/coinbase"
+	"github.com/skip-mev/slinky/providers/apis/coingecko"
+	apihandlers "github.com/skip-mev/slinky/providers/base/api/handlers"
+	"github.com/skip-mev/slinky/providers/base/api/metrics"
+	"github.com/skip-mev/slinky/providers/static"
+	"github.com/skip-mev/slinky/providers/types/factory"
+	oracletypes "github.com/skip-mev/slinky/x/oracle/types"
+	"go.uber.org/zap"
+)
+
+// OracleAPIQueryHandlerFactory returns a sample implementation of the API query handler factory.
+func OracleAPIQueryHandlerFactory() factory.APIQueryHandlerFactory[oracletypes.CurrencyPair, *big.Int] {
+	return func(logger *zap.Logger, cfg config.ProviderConfig, metrics metrics.APIMetrics) (apihandlers.APIQueryHandler[oracletypes.CurrencyPair, *big.Int], error) {
```

Comment:
This is all copy pasted from the previous implementation but now moved into a separate function.